pkg/accounts/model/account.ts

```diff
@@ -14,6 +16,48 @@ export type AccountStatus = 'active' | 'notActivated';
 export type AccountFrozen = 'frozen' | 'normal';
 export type AccountSilenced = 'silenced' | 'normal';
 
+export const AccountNameSchema = z
+  .string()
+  .refine((s) => {
+    const parts = s.split('@', 3);
+
+    // check. @ 区切りで 3 つの文字列に区切ることが出来る
+    if (!((p): p is [string, string, string] => p.length === 3)(parts)) {
+      return false;
+    }
+
+    const [head, name, domain] = parts;
+
+    // check. 最初の文字は @, 最初の @ の手前は空文字
+    if (head.length !== 0) {
+      return false;
+    }
+
+    // check. 名前は a-z A-Z 0-9 - _ . のみ許容
+    //        但し 1 文字以上, 最初の文字は記号非許容
+    if (!/^[a-zA-Z0-9][\w\-.]*$/.test(name)) {
+      return false;
+    }
+
+    // ref. RFC1035 https://www.rfc-editor.org/rfc/rfc1035#page-8
+    //
+    // check. ドメイン名は RFC1035 より "<subdomain>" を参照, 空白は許容しない
+    //        ここでは文字種の検証のみ
+    if (!/^[a-zA-Z0-9\-.]+$/.test(domain)) {
+      return false;
+    }
+
+    // check. RFC1035 より "<label>" を参照
+    for (const label of domain.split('.')) {
+      if (!/^[a-zA-Z](?:.*[a-zA-Z0-9])?$/.test(label)) {
```

Comment:
```suggestion
      if (!/^[a-zA-Z](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?$/.test(label)) {
```
RFC を読む限りだとこういう定義になっているかと

Reply:
```ts
    // ref. RFC1035 https://www.rfc-editor.org/rfc/rfc1035#page-8
    //
    // check. ドメイン名は RFC1035 より "<subdomain>" を参照, 空白は許容しない
    //        ここでは文字種の検証のみ
    if (!/^[a-zA-Z0-9\-.]+$/.test(domain)) {
      return false;
    }
```
ここで文字種に関する識別は済ませているつもりですが, 2 重チェックにしようと思います

Reply:
2a453ca9f331a6df1cabae31eac2d2c242cec45c で対応しました.

Reply:
この修正だとドメイン名の `-` が末尾に含まれるケースで間違って通過しそうです. `@name@example-.com` などの場合でも期待通り拒否されるでしょうか?

Reply:
432126b371ea40dd0b357824e08c192a8a914093 でテストケースを追加しました.
また, それについては拒否されないので a57f612090ca7da8eb3cda0e756565290895a462 で対応しましたが, いかがでしょうか.